File: android-app/designsystem/src/main/java/tv/bayit/plus/designsystem/component/SleepTimerBanner.kt
package tv.bayit.plus.designsystem.component

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.animation.slideInVertically
import androidx.compose.animation.slideOutVertically
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Bedtime
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import tv.bayit.plus.designsystem.i18n.bayitString
import tv.bayit.plus.designsystem.modifier.glassMorphism
import tv.bayit.plus.designsystem.theme.DesignTokens

private const val EXTEND_MINUTES = 5

/**
 * Compact banner overlay showing sleep timer countdown.
 * Displays moon icon, remaining time, extend button, and cancel control.
 * Uses [AnimatedVisibility] for smooth enter/exit transitions.
 */
@Composable
fun SleepTimerBanner(
    isVisible: Boolean,
    remainingSeconds: Int,
    onExtend: (Int) -> Unit,
    onCancel: () -> Unit,
    modifier: Modifier = Modifier,
) {
    AnimatedVisibility(
        visible = isVisible,
        enter = slideInVertically(initialOffsetY = { -it }) + fadeIn(),
        exit = slideOutVertically(targetOffsetY = { -it }) + fadeOut(),
    ) {
        Row(
            modifier = modifier
                .fillMaxWidth()
                .glassMorphism()
                .padding(
                    horizontal = DesignTokens.Spacing.md,
                    vertical = DesignTokens.Spacing.xs,
                ),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween,
        ) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(DesignTokens.Spacing.xs),
            ) {
                Icon(
                    imageVector = Icons.Default.Bedtime,
                    contentDescription = null,
                    tint = DesignTokens.Colors.Primary.light,
                    modifier = Modifier.padding(end = DesignTokens.Spacing.xxs),
                )
                Text(
                    text = bayitString(
                        "player.sleepTimer.remaining",
                        mapOf("time" to formatCountdown(remainingSeconds)),
                    ),
                    color = DesignTokens.Colors.Primary.light,
                    fontSize = DesignTokens.FontSize.sm,
                    fontWeight = FontWeight.Medium,
                )
            }

            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(DesignTokens.Spacing.xs),
            ) {
                GlassButton(
                    text = bayitString(
                        "player.sleepTimer.extend",
                        mapOf("minutes" to EXTEND_MINUTES.toString()),
                    ),
                    onClick = { onExtend(EXTEND_MINUTES) },
                    isPrimary = false,
                )
                IconButton(onClick = onCancel) {
                    Icon(
                        imageVector = Icons.Default.Close,
                        contentDescription = bayitString("player.sleepTimer.cancel"),
                        tint = DesignTokens.Colors.Text.muted,
                    )
                }
            }
        }
    }
}

private fun formatCountdown(totalSeconds: Int): String {
    val minutes = totalSeconds / 60
    val seconds = totalSeconds % 60
    return "${minutes.toString().padStart(2, '0')}:${seconds.toString().padStart(2, '0')}"
}

File: ios-app/BayitPlusApp/Views/Player/SleepTimerBanner.swift
import BayitDesignSystem
import SwiftUI

/// Compact countdown banner displaying remaining sleep timer time.
/// Shows moon icon, formatted countdown, extend button, and cancel control.
/// Glass-styled to match `MiniAudioPlayerBar`.
struct SleepTimerBanner: View {
    let remainingSeconds: Int
    let onExtend: (Int) -> Void
    let onCancel: () -> Void

    @State private var localization = LocalizationManager()

    private static let extendMinutes = 5

    var body: some View {
        if remainingSeconds > 0 {
            HStack {
                HStack(spacing: DesignTokens.Spacing.xs) {
                    Image(systemName: "moon.zzz.fill")
                        .font(.system(size: DesignTokens.FontSize.sm))
                        .foregroundColor(DesignTokens.Primary.default)

                    Text(localization.t("player.sleepTimer.remaining", ["time": formatCountdown(remainingSeconds)]))
                        .font(.system(size: DesignTokens.FontSize.sm, weight: .medium))
                        .foregroundColor(DesignTokens.Primary.default)
                        .monospacedDigit()
                }

                Spacer()

                HStack(spacing: DesignTokens.Spacing.sm) {
                    Button {
                        onExtend(Self.extendMinutes)
                    } label: {
                        Text(localization.t("player.sleepTimer.extend", ["minutes": "\(Self.extendMinutes)"]))
                            .font(.system(size: DesignTokens.FontSize.xs, weight: .semibold))
                            .foregroundColor(DesignTokens.Primary.default)
                            .padding(.horizontal, DesignTokens.Spacing.sm)
                            .padding(.vertical, 2)
                            .background(DesignTokens.Primary.default.opacity(0.15))
                            .clipShape(RoundedRectangle(cornerRadius: DesignTokens.Radius.sm))
                    }
                    .accessibilityLabel(localization.t("player.sleepTimer.extend", ["minutes": "\(Self.extendMinutes)"]))

                    Button {
                        onCancel()
                    } label: {
                        Image(systemName: "xmark")
                            .font(.system(size: 10, weight: .semibold))
                            .foregroundColor(DesignTokens.Text.muted)
                            .frame(width: 20, height: 20)
                    }
                    .accessibilityLabel(localization.t("player.sleepTimer.cancel"))
                }
            }
            .padding(.horizontal, DesignTokens.Spacing.md)
            .padding(.vertical, DesignTokens.Spacing.xs)
            .glassBackground()
            .clipShape(RoundedRectangle(cornerRadius: DesignTokens.Radius.md))
            .overlay(
                RoundedRectangle(cornerRadius: DesignTokens.Radius.md)
                    .stroke(DesignTokens.Primary.default.opacity(0.2), lineWidth: 1)
            )
        }
    }

    private func formatCountdown(_ totalSeconds: Int) -> String {
        let minutes = totalSeconds / 60
        let seconds = totalSeconds % 60
        return String(format: "%02d:%02d", minutes, seconds)
    }
}
